Dockerfile.updater-core

```diff
@@ -90,14 +90,14 @@ WORKDIR $DEPENDABOT_HOME/dependabot-updater
 
 # Install Ruby from official Docker image
 # When bumping Ruby minor, need to also add the previous version to `bundler/helpers/v{1,2}/monkey_patches/definition_ruby_version_patch.rb`
-COPY --from=ruby:3.1.4-bullseye --chown=dependabot:dependabot /usr/local /usr/local
+COPY --from=ruby:3.1.4-bookworm --chown=dependabot:dependabot /usr/local /usr/local
 
 # When bumping Bundler, need to also regenerate `updater/Gemfile.lock` via `bundle update --lock --bundler`
 # Generally simplest to match the bundler version to the one that comes by default with whatever Ruby version we install.
 # This way other projects that import this library don't have to futz around with installing new / unexpected bundler versions.
 ARG BUNDLER_V2_VERSION=2.4.17
 
-# We had to explicitly bump this as the bundled version `0.2.2` in ubuntu 20.04 has a bug.
+# We had to explicitly bump this as the bundled version `0.2.2` in ubuntu 22.04 has a bug.
```

Comment:
Does the new version of Ubuntu handle the yaml parsing now? And can we therefore undo this [PR](https://github.com/dependabot/dependabot-core/pull/7112/files)

Reply:
I wish we could, but we'll need Ubuntu `23.04` before we get a new enough version:
 * https://packages.ubuntu.com/search?suite=default&section=all&arch=any&keywords=libyaml-0-2&searchon=names

We have a :dependabot: PR to bump to `23.04`, but probably should land this PR first, and then also have a conversation about do we even want to bump to non-LTS versions:
 * https://github.com/dependabot/dependabot-core/pull/6561